File: 1-bit-and-2-bit-characters/1-bit_and_2-bit_characters.js
/**
 * @param {number[]} bits
 * @return {boolean}
 */
var isOneBitCharacter = function(bits) {
    var i = 0;
    while (i < bits.length) {
        if (i == bits.length - 1) {
            return true;
        } else if (bits[i] == 1) {
            i += 2;
        } else {
            i += 1;
        }
    }
    return false;
};

File: 1-bit-and-2-bit-characters/1-bit_and_2-bit_characters.go
func isOneBitCharacter(bits []int) bool {
    var i int
    for i < len(bits) {
        if i == len(bits) - 1 {
            return true
        } else if bits[i] == 1 {
            i += 2
        } else {
            i += 1
        }
    }
    return false
}